magento/pwa-studio
# [Cart v2] ProductListing Component

## Description

Within the Cart Page component, if a product has been added to cart it will display with necessary information and actions for a shopper to view/edit/delete. 

 

Requirements: 

Adhere to design specs/flows provided
Empty cart should display appropriate messaging
"Sign in" button opens sign in modal that returns back to cart once user is auth'ed
User can update quantity using increment/decrement functionality
Kebab menu should reuse options from existing Venia mini-cart
Open Questions: 

What product attributes should be listed in the view? For now, it's just size/color in the mockup. As discussed in grooming, all product options should be displayed for now. 

## Related Issue





* [[PWA-238](https://jira.corp.magento.com/browse/PWA-238)] Product Listing (Cart)

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


**Data fetch policy I used for ease of development has been removed so its not accidentally left in once all cart features are completed. You'll need to clear your Apollo cache if you make any cart mutations outside of this view (ie. Add to Cart).**
1. Navigate to `/cart` route, verify empty cart message is shown
2. Add Products to your cart, both simple and configurable
3. Go back to `/cart` route, verify details are displayed and aligned with mocks
4. Switch to mobile view and verify layout collapses and adheres to mocks

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
